Assets/Scripts/ROS Scripts/ROS Connections v2/Drones/Matrice_ROSDroneConnection.cs

```diff
@@ -0,0 +1,846 @@
+        return velocity;
+    }
+
+    /// <summary>
+    /// Current angles of attached gimble
+    /// </summary>
+    /// <returns></returns>
+    public Vector3 GetGimbleJointAngles()
+    {
+        return gimble_joint_angles;
+    }
+
+    /// <summary>
+    /// Strength of GPS connection
+    /// </summary>
+    /// <returns></returns>
+    public float GetGPSHealth()
+    {
+        return gps_health;
+    }
+
+    /// <summary>
+    /// Home Latitude of the drone
+    /// </summary>
+    /// <returns></returns>
+    public double GetHomeLat()
+    {
+        return droneHomeLat;
+    }
+
+    /// <summary>
+    /// Home Longitude of the drone
+    /// </summary>
+    /// <returns></returns>
+    public double GetHomeLong()
+    {
+        return droneHomeLong;
+    }
+
+    /// ROSTopicSubscriber Interface methods
+
+    /// <summary>
+    /// Parse received message from drone based on topic and perform required action
+    /// </summary>
+    /// <param name="topic"></param>
+    /// <param name="raw_msg"></param>
+    /// <param name="parsed"></param>
+    /// <returns></returns>
+    public ROSBridgeMsg OnReceiveMessage(string topic, JSONNode raw_msg, ROSBridgeMsg parsed = null)
+    {
+        ROSBridgeMsg result = null;
+        // Writing all code in here for now. May need to move out to separate handler functions when it gets too unwieldy.
+        switch (topic)
+        {
+            case "/dji_sdk/attitude":
+                QuaternionMsg attitudeMsg = (parsed == null) ? new QuaternionMsg(raw_msg["quaternion"]) : (QuaternionMsg)parsed;
+                attitude = offset * (new Quaternion(attitudeMsg.GetX(), attitudeMsg.GetY(), attitudeMsg.GetZ(), attitudeMsg.GetW()));
+                result = attitudeMsg;
+                break;
+            case "/dji_sdk/battery_state":
+                battery_state = (parsed == null) ? new BatteryStateMsg(raw_msg) : (BatteryStateMsg)parsed;
+                result = battery_state;
+                break;
+            case "/dji_sdk/flight_status":
+                flight_status = (FlightStatus)(new UInt8Msg(raw_msg)).GetData();
+                break;
+            case "/dji_sdk/gimbal_angle":
+                Vector3Msg gimbleAngleMsg = (parsed == null) ? new Vector3Msg(raw_msg["vector"]) : (Vector3Msg)parsed;
+                gimble_joint_angles = new Vector3((float)gimbleAngleMsg.GetX(), (float)gimbleAngleMsg.GetY(), (float)gimbleAngleMsg.GetZ());
+                result = gimbleAngleMsg;
+                break;
+            case "/dji_sdk/gps_health":
+                gps_health = (parsed == null) ? (new UInt8Msg(raw_msg)).GetData() : ((UInt8Msg)parsed).GetData();
+                break;
+            case "/dji_sdk/gps_position":
+                gps_position = (parsed == null) ? new NavSatFixMsg(raw_msg) : (NavSatFixMsg)parsed;
+                result = gps_position;
+
+                // Peru 6/14/20: Set drone home latitude and longitutde as first message from drone gps position.
```

Comment:
what does this comment mean?

Reply:
also why the date?

Reply:
changed to todo, no idea why I have a habit of putting dates.